.builder/actions/aws_crt_python.py

```diff
@@ -233,9 +14,15 @@ def run(self, env):
         args = parser.parse_known_args(env.args.args)[0]
         python = args.python if args.python else PYTHON_DEFAULT
 
+        # Enable S3 tests
+        env.shell.setenv('AWS_TEST_S3', '1')
+
         actions = [
+            # We force install pip because some containers cannot uninstall and upgrade pip, but we need a newer
+            # pip for setuptools. Not ideal, but gets the job done.
+            [python, '-m', 'pip', 'install', '--upgrade', '--ignore-installed', 'pip'],
```

Comment:
The issue I was having with pip was: https://github.com/awslabs/aws-crt-python/actions/runs/4876483632/jobs/8700041335#step:2:243

I agree that `--ignore-installed` is not ideal though, but because it is not installed with a RECORD file (meaning it was not installed with Python pip...?) it cannot uninstall it and therefore cannot install a new version. At first I skipped updating pip entirely, but pip needs to be updated for the setup tools to work.

Reply:
I think I have fixed it by trying to install pip a number of different ways back to back until it succeeds, starting with `python -m pip install --upgrade pip` and ending with the `--ignore_installed` only if none of the other methods do not work AND only if running in GitHub actions by checking for the `GITHUB_ACTIONS` environment variable`.